src/PowerSystem/power_network.jl

```diff
@@ -78,6 +66,25 @@ struct PowerNetwork
     end
 end
 
+get(pf::PowerNetwork, ::NumberOfBuses) = pf.nbus
+get(pf::PowerNetwork, ::NumberOfLines) = size(pf.data["branch"], 1)
+get(pf::PowerNetwork, ::NumberOfGenerators) = pf.ngen
+get(pf::PowerNetwork, ::NumberOfPVBuses) = length(pf.pv)
+get(pf::PowerNetwork, ::NumberOfPQBuses) = length(pf.pq)
+get(pf::PowerNetwork, ::NumberOfSlackBuses) = length(pf.ref)
+
+function get(pf::PowerNetwork, ::GeneratorIndexes)
```

Comment:
At some point, we may want to leverage this function to support multiple generators by bus.